content/en/docs/devops-user-guide/how-to-use/use-pipeline-templates.md

```diff
@@ -0,0 +1,280 @@
+
+2. In the dialog that appears, name it `ci-pipeline` and click **Next**.
+
+   ![ci-pipeline](/images/docs/devops-user-guide/using-devops/use-pipeline-templates/ci-pipeline.png)
+
+3. On the **Advanced Settings** tab, click **Add Parameter** to add three string parameters as follows. These parameters will be used in the Docker commands in the pipeline. Click **Create** when you finish adding.
+
+   ![add-parameter](/images/docs/devops-user-guide/using-devops/use-pipeline-templates/add-parameter.png)
+
+   | Parameter Type | Name                | Value       | Description                                                  |
+   | -------------- | ------------------- | ----------- | ------------------------------------------------------------ |
+   | String         | REGISTRY            | `docker.io` | This is the image registry address. This example uses `docker.io`. |
+   | String         | DOCKERHUB_NAMESPACE | Docker ID   | You Docker Hub account or the organization name under the account. |
+   | String         | APP_NAME            | `ci-sample` | The app name.                                                |
+
+4. The pipeline created will appear in the list.
+
+   ![ci-pipeline-created](/images/docs/devops-user-guide/using-devops/use-pipeline-templates/ci-pipeline-created.png)
+
+### Step 3: Edit the pipeline
+
+Click the pipeline to go to its detail page, and then click **Edit Pipeline** under the **Pipeline** tab. In the dialog that appears, click the pipeline template for continuous integration.
+
+![edit-ci-pipeline](/images/docs/devops-user-guide/using-devops/use-pipeline-templates/edit-ci-pipeline.png)
+
+![ci-template](/images/docs/devops-user-guide/using-devops/use-pipeline-templates/ci-template.png)
+
+On the graphical editing panel, there are two predefined stages with built-in steps. You can customize these steps based on your needs.
+
+![ci-stages](/images/docs/devops-user-guide/using-devops/use-pipeline-templates/ci-stages.png)
+
+For example, you can select **maven** from the **label** drop-down list.
+
+![select-maven](/images/docs/devops-user-guide/using-devops/use-pipeline-templates/select-maven.png)
+
+#### Stage 1: clone code
+
+1. Click **clone code** to select this stage, and you can see the built-in steps shown on the right. For the step of **container**, click the dustbin icon on the right of it.
+
+   ![delete-container-step](/images/docs/devops-user-guide/using-devops/use-pipeline-templates/delete-container-step.png)
+
+2. Click **Add Step**.
+
+   ![click-add-step](/images/docs/devops-user-guide/using-devops/use-pipeline-templates/click-add-step.png)
+
+3. Select **git** from the list on the right as the example code is pulled from GitHub. In the dialog that appears, enter the GitHub repository address `https://github.com/Felixnoo/devops-java-sample.git`. Note that this is an example and you need to use your own repository address. Click **OK** to finish.
```

Comment:
This is a personal github repository. Do we have this repository in KubeSphere? Usually, we need to give a public KubeSphere github repository and tell readers to fork it.

Reply:
The pipeline template has a built-in step that uses the SVN repository and Shaowen told me to find one available.
However, I can't find any. So I used the GitHub repository mentioned in the [graphical pipeline document](https://kubesphere.io/docs/devops-user-guide/how-to-use/create-a-pipeline-using-graphical-editing-panel/) and added a few new files after forking in order to make the pipeline templates work. 
Do I need to add the contents about the changes I made? I'm starting to think that this document is getting too complicated and cannot make users feel the convenience of pipeline templates...

Reply:
Yes. I reckon this guide should be a concept walk-through guide instead of a how-to guide. Readers do not need to know how to use the graphical editing panel which is already covered in another guide.